views/containers/Camera.jsx

```diff
@@ -1,16 +1,58 @@
+import React from 'react';
 import { connect } from 'react-redux';
-import MovingCamera from '../components/MovingCamera';
+import { cameraRotation } from '../actions/Camera';
+import keyEmitter from '../KeyEventEmitter';
+const ReactTHREE = require('react-three');
+const PerspectiveCamera = ReactTHREE.PerspectiveCamera;
+
+const keys = ['J','K','L','I','ENTER'];
+
+class MovingCamera extends React.Component {
+  componentDidMount = () => {
+      this.props.onMount()
+  };
+
+  componentWillUnmount = () => {
+      this.props.onUnmount()
+  };
+
+  render() {
+    const cameraConfig = {
+      ...this.props.config,
+      aspect: this.props.aspect,
+    };
+
+    return (
+      <PerspectiveCamera name="maincamera" {...cameraConfig} />
+    );
+  }
+}
+
+MovingCamera.propTypes = {
+  onUnmount: React.PropTypes.func.isRequired,
+  onMount: React.PropTypes.func.isRequired,
+  config: React.PropTypes.shape({
+    fov: React.PropTypes.number.isRequired,
+    near: React.PropTypes.number.isRequired,
+    far: React.PropTypes.number.isRequired,
+    position: React.PropTypes.object.isRequired,
+    lookat: React.PropTypes.object.isRequired,
+  }).isRequired,
+  aspect: React.PropTypes.number.isRequired,
+};
 
 const mapStateToProps = (state) => {
   return {
-    config: state.Camera.config,
+    config: state.CharacterCamera.cameraConfig,
     aspect: state.Browser.size.width / state.Browser.size.height,
   };
 };
 
 const mapDispatchToProps = (dispatch) => {
+  const onKeyPressed = (event)=>{dispatch(cameraRotation(event))};
   return {
-    dispatch: dispatch,
+    onMount: () => { keyEmitter.on(keys,onKeyPressed)},
+    onUnmount: () => { keyEmitter.off(keys,onKeyPressed)}
```

Comment:
We can avoid this entirely if we create a separate component to handle all the controls.

```javascript
import React from 'react';
import { connect } from 'react-redux';
import keyEmitter from '../KeyEventEmitter';
const ReactTHREE = require('react-three');

class Controllers extends React.Component {
  componentDidMount = () => {
    // subscribe to keypress.
  };

  componentWillUnmount = () => {
    // un-subscribe from keypress
  };

  handleKeyPress = (event) => {
    switch (event.keyChar) {      
      case KEYS.W:
        return dispatch(characterMoveForward());
      case KEYS.A:
        return dispatch(characterMoveLeft());
      case KEYS.S:
        return dispatch(characterMoveDown());
      case KEYS.D:
        return dispatch(characterMoveRight());

      // TODO: Add remaining key-binginds
    }
  }

  render() {
    return false;
  }
}

const Controllers = connect(
  mapStateToProps,
  mapDispatchToProps
)(MovingCamera);

export default Camera;
```

And you use this component at root-level.